src/page-manager/wrapper.js

```diff
@@ -0,0 +1,173 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * Customize a route to be display inside a wrapper component like a toast
+ * or a dialog. This will also handle routing and dependencies loading
+ *
+ * @see  PageManager route option
+ */
+define([
+            'jquery',
+            'ju-shared/class'
+        ],
+        function(
+            $,
+            Class
+        ) {
+
+    'use strict';
+
+    var RouteWrapper = Class.extend({
+        /**
+         * Constructor
+         *
+         * @param  {object} controllerInfo  route defintion
+         * @param  {object} controllerStack PageManager controllers stack
+         */
+        init : function(controllerInfo, controllerStack) {
+            // Store controllerInfo reference
+            this.controllerInfo = controllerInfo || {};
+            this.hasWrapper = !!this.controllerInfo.controllerWrapper,
+            this.controllerWrapper = this.hasWrapper ? this.controllerInfo.controllerWrapper : {};
+
+            // Ease access to the number of controllers in stack
+            var controllerStackLength = controllerStack.length;
+
+            // Flags : requested controller can be root or stacked
+            // By default, a requested controller is root so it is not stacked
+            this.isRootController = controllerStackLength === 0;
+            this.doStackController = !this.isRootController && this.hasWrapper;
+
+            // Store reference to current controller (not necessarily the first in the stack)
+            this.currentController = this.doStackController ?
+                controllerStack[controllerStackLength - 1] : null;
+        },
+        /**
+         * Prepare requested controller info based on how it is requested:
+         * - Directly: display without wrapper (view)
+         * - After: display wrapper (overlay)
+         *
+         * @return {Object} original or modified clone of controller info
+         */
+        prepareControllerInfo : function() {
```

Comment:
I think we should change this method to avoid the side effect, as we're modifying a shared reference to an object and that's hard to maintain.

To me, it'll be easier to just return the additional options to be added to the original `controllerInfo` object, and extend it on the `PageManager`, which is the owner of that object reference